### ISSUE TITLE
Authentication Issues

### ISSUE THREAD

**Issue by sambasile** [pantheon-systems/localdev-issues#63]
Clean setup of the latest Localdev 0.10.0.  Accepts machine token and authenticates without issue during installation.  But attempting to pull a local dev fails.

Attempting to login via terminus...
 [notice] Logging in via machine token.
 [warning] HTTPS request failed with error cURL error 77: error setting certificate verify locations:
  CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs (see https://curl.haxx.se/libcurl/c/libcurl-errors.html). Retrying in 146 milliseconds..
 [warning] HTTPS request failed with error cURL error 77: error setting certificate verify locations:
  CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs (see https://curl.haxx.se/libcurl/c/libcurl-errors.html). Retrying in 291 milliseconds..
 [warning] HTTPS request failed with error cURL error 77: error setting certificate verify locations:
  CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs (see https://curl.haxx.se/libcurl/c/libcurl-errors.html). Retrying in 468 milliseconds..
 [warning] HTTPS request failed with error cURL error 77: error setting certificate verify locations:
  CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs (see https://curl.haxx.se/libcurl/c/libcurl-errors.html). Retrying in 853 milliseconds..
 [warning] HTTPS request failed with error cURL error 77: error setting certificate verify locations:
  CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs (see https://curl.haxx.se/libcurl/c/libcurl-errors.html). Retrying in 1669 milliseconds..
 [error]  HTTPS request failed with error cURL error 77: error setting certificate verify locations:
  CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs (see https://curl.haxx.se/libcurl/c/libcurl-errors.html). Maximum retry attempts reached. 
 [error]  Could not find a saved token identified by . 
 [error]  Please visit the dashboard to generate a machine token:https://dashboard.pantheon.io/machine-token/create/_tokenvaluewashere__

**Comment by joe-westcott** [comment]
Hi! I'm having the same problem. This means that Localdev is no longer usable for me, as of version 0.10.0, which is a blocker.

Is there a way to download the previous 0.9.x version for now?

I contacted Pantheon support, but they were unable to help with this error, and they didn't know where to find an older download. They advised me to report an issue here, and I see that sambasile already has.

**Comment by joe-westcott** [comment]
If it helps, this bug only affects users who pull envs that aren't already pulled in Localdev version 0.10.0, I guess because the authentication function only runs in certain situations.

So if you pull your dev environment in Localdev 0.9.3, then update to 0.10.0, and then pull dev again, then the pull will work -- you will _not_ experience this bug.

But if you perform either of the following actions, then Localdev 0.10.0 will fail:

1. Run an older version of Localdev, then destroy an env. Upgrade to Localdev 0.10.0, and pull the env that you destroyed. This will fail.

2. Completely delete Localdev and its folders, then install 0.10.0, then attempt to pull an env. This will fail.

**Comment by dustinleblanc** [comment]
@joe-westcott I've been working quite a bit to reproduce this issue, and it seems less straight-forward than outlined above. I can run from a completely fresh install and just rerun the initial site pull over and over again until it works. The issue is related to the machine token not being set and we're working on tracking down the cause.

Thanks!

**Comment by joe-westcott** [comment]
Thanks, Dustin. Maybe it’s only for machines where 0.9.3 was installed first?

I followed Pantheon’s instructions to delete the app and delete the .Localdev folder before I re-installed Localdev 0.10.0. But Pantheon also told me to delete a folder that didn’t seem to exist. So maybe I didn’t have the correct instructions to completely remove Localdev 0.9.3, in which case, maybe something is still in my machine from 0.9.3, and interfering here.

**Comment by joe-westcott** [comment]
If it help, here were Pantheon’s instructions to uninstall:

```
1) Reinstall latest docker
2) Delete Localdev from you Applications and empty the trash
3) Delete ~/.pantheonlocaldev folder (also please try having a backup)
4) Delete the ~/.Localdev folder (also please try having a backup)
5) Reinstall the latest Localdev from here https://pantheon-localdev.s3.amazonaws.com/localdev-stable.dmg

If it will not work, we may need to open up an issue here https://github.com/pantheon-systems/localdev-issues/issues then adding your OS version, docker and localdev version along with the error logs /.pantheonlocaldev/logs/error.log
```

And here were the problems with those instructions:

1. ~/.Localdev did not exist, and it still doesn't exist after I follow these steps to reinstall.

2. ~/.pantheonlocaldev/logs/error.log does not exist. Instead, I see these four files:

i. childcareaware_missingcopy-error.log
ii. childcareaware_missingcopy.log
iii. lando-error.log
iv. lando.log

**Comment by dustinleblanc** [comment]
@joe-westcott I think the issue is not dependent on a previous install, I am able to reproduce with a completely clean system running from source. replace `~/.Localdev` with `~/Localdev` and you'll have the right folder 😄 

That folder houses the default locations for your applications/sites themselves where as `~/.pantheonlocaldev` is the location for Localdev's config/cache files, etc.

We are working on this during this week as well

**Comment by pirog** [comment]
@dustinleblanc i merged your fix for this into `master` and am rolling a `0.10.1` release. Def encourage everyone who had issues with this to try that out.

@dustinleblanc might also be worth following up with Jurgen to tell him about this as well.

**Comment by dustinleblanc** [comment]
@sambasile @joe-westcott let us know if 0.10.1 fixes it for you folks

**Comment by joe-westcott** [comment]
I updated to 0.10.1, and it works!

Docker is still set to version 2.2.0.4 after I upgraded from Localdev 0.9.3 to 0.10.1. Is that correct?

Previously, when I upgraded deom Localdev 0.9.3 to 0.10.0, I thought that it updated Docker.

**Comment by joe-westcott** [comment]
Sorry, I wrote too soon. Localdev 0.10.1 fixes the auth issues but then there's a later problem with pulling down sites.

I'll see whether a different Docker version might fix this.

**Comment by joe-westcott** [comment]
Okay, manually updating Docker solved the problem where I couldn't pull down environments to my local machine. All good now!

I'd suggest a feature to have upgrade process check that the correct Docker version is installed, to help other people who'll get frustrated when the automated upgrade breaks things.

But this workaround is great for now! Thank you for releasing this new version of Localdev, and all your work to fix the auth problem in version 0.10.0.

**Comment by pirog** [comment]
@joe-westcott localdev already has this feature but we do not upgrade docker unless localdev requires an upgrade to docker. as far as we know version 2.2.0.4 should be sufficient so i would be great to know what didnt work with yours. 

my suspicion is that the problem you ran into is with docker being out of storage and not necessarily with the version different itself.
